docs/dev/request-lifecycle.md

```diff
@@ -0,0 +1,251 @@
+# Solve Request Lifecycle
+
+Buildkit solves build graphs to find the final result. By default, nothing will
+be exported to the client, but requests can be made after solving the graph to
+export results to external destinations (like the client’s filesystem).
+
+A solve request goes through the following:
+
+1. Client makes a solve request and sends it to buildkitd over gRPC. The
+   request may either include a LLB definition, or the name of a frontend (must
+   be `dockerfile.v0` or `gateway.v0`), but it must not be both.
+2. Buildkitd receives the solve request with the Controller. The controller is
+   registered as the ControlServer gRPC service.
+3. The controller passes it down to the LLB solver, which will create a job for
+   this request. It will also create a FrontendLLBBridge, that provides a
+   solving interface over the job object.
+4. The request is processed:
+   - If the request is definition-based, it will simply build the definition.
+   - If the request is frontend-based, it will run the frontend over the
+     gateway while passing it a reference to the FrontendLLBBridge. Frontends
+     must return a result for the solve request, but they may also issue solve
+     requests themselves to the bridge.
+5. The results are plumbed back to the client, and the temporary job and bridge
+   are discarded.
+
+```plantuml
+@startuml
```

Comment:
You could render uml files with http://www.plantuml.com and markdown:

```
![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://gist.githubusercontent.com/crazy-max/2397697b7237af038c1faaa790703c0e/raw/1bcf6404ed00aca4c0651c975fa6b1cc6e54bd25/test.uml)
```

![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://gist.githubusercontent.com/crazy-max/2397697b7237af038c1faaa790703c0e/raw/1bcf6404ed00aca4c0651c975fa6b1cc6e54bd25/test.uml)

Reply:
Have resolved this by converting the diagrams into mermaid-js, which is natively supported on github pages :+1:

Colours in mermaid diagrams aren't supported, as well as grouped actors, but `FIXME`s have been added inline, for when/if mermaid supports these.